src/main/java/view/GameRoomView.java

```diff
@@ -0,0 +1,161 @@
+package view;
+
+import java.awt.*;
+import java.awt.event.ActionListener;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+
+import javax.swing.*;
+
+import interface_adapter.set_difficulty.SetDifficultyController;
+import interface_adapter.set_difficulty.SetDifficultyState;
+import interface_adapter.set_difficulty.SetDifficultyViewModel;
+
+/**
+ * The View for when the user is setting up a game.
+ */
+public class GameRoomView extends JPanel implements PropertyChangeListener {
```

Comment:
Maybe we can have a call about this; I don't think we can have a "GameRoomView". Views are very concrete low level details, this would have to be an explicit view like "BlackjackView" or "SomeOtherGame" view and in that it would have very specific hard dependencies on use cases that are specific to that view, like hitting or passing for Blackjack. 

I'm working on a BlackjackView right now, will see what I can merge from this into that.

Reply:
Okay, you're definitely right. I named it so badly, should've been "blackjackview".